data/transactions/transaction.go

```diff
@@ -365,6 +365,10 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 			return fmt.Errorf("tx.ForeignApps too long, max number of foreign apps is %d", proto.MaxAppTxnForeignApps)
 		}
 
+		if len(tx.ForeignAssets) > proto.MaxAppTxnForeignAssets {
+			return fmt.Errorf("tx.ForeignAssets too long, max number of foreign assets is %d", proto.MaxAppTxnForeignAssets)
```

Comment:
I think that the error messages here should be reformatted, as they would typically be presented as a goal errors strings. For this particular error, I think that the following would be a better choice:
```
return fmt.Errorf("the ForeignAssets array length is too long, containing %d assets IDs whereas the max number of supported foreign assets is %d", len(tx.ForeignAssets), proto.MaxAppTxnForeignAssets)
```

Reply:
Fair point, I'd like to update them all at once in a separate PR, however. I'll make a ticket so I don't forget.